test/types/function.civet

```diff
@@ -74,6 +74,17 @@ describe "[TS] function", ->
     }
   """
 
+  testCase """
+    longhand optional return
+    ---
+    function x(a: number, b?: number) ?: number
```

Comment:
I'm not as happy with this syntax as with the `let x?: T` syntax.

`function x(a: number, b?: number): number?` seems preferable. TypeScript even seems to be indirectly pushing in that direction as well:
![image](https://github.com/DanielXMoore/Civet/assets/18894/6a4c9990-73a8-46c6-9f54-f6e308ac7425)

I'm happy to merge this to gain more practical usage and experimentation with the expectation that we may change or deprecate as we learn more and consolidate with #144

Reply:
I think I agree with you. But I do think this syntax is symmetric, and maybe the user would prefer one consistent style or another. 

#144 offers an alternative syntax for all optional types, so presumably someone could switch wholesale to `?` next to the type. Before or after depends on what we think is best. Also this depends on whether we think `null` should be part of `?` as a type operator.

My thought was, while we figure out #144, this syntax seems unambiguous so we can do it now. (That said, eager to hear your thoughts on which way #144 should go.)